src/window.rs

```diff
@@ -1516,3 +1527,21 @@ impl Default for WindowLevel {
         Self::Normal
     }
 }
+
+/// IME purposes.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[non_exhaustive]
+pub enum ImePurpose {
+    /// default input
```

Comment:
Maybe something like that?

```suggestion
    /// No special hints for the IME.
```

Reply:
And also likely add a note that it's a default.